lib/game_logic.rb

```diff
@@ -0,0 +1,109 @@
+module WinNumbers
+  WIN_COMBINATIONS = [
+    [0, 1, 2],
+    [3, 4, 5],
+    [6, 7, 8],
+    [0, 3, 6],
+    [1, 4, 7],
+    [2, 5, 8],
+    [6, 4, 2],
+    [0, 4, 8]
+  ].freeze
+end
+class GameLogic
+  def valid_move?(input)
+    input.between?(0, 9) && !position_taken?(input)
+  end
+
+  def game_on?
+    turn_counts < 9 && !win? ? true : false
+  end
+
+  def draw?
+    !win? && game_full?
+  end
+
+  def game_over?
+    draw? || win? || game_full
+  end
+
+  def winner
+    player = ''
+    player == win? ? @board[winner.first] : @players
+  end
+end
+class TicTacToe < GameLogic
+  attr_reader :board
+  include WinNumbers
+  def initialize
+    @board = [' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ']
+  end
+
+  def display_board
+    puts " #{@board[0]} | #{@board[1]} | #{@board[2]} "
+    puts ' ----------- '
+    puts " #{@board[3]} | #{@board[4]} | #{@board[5]} "
+    puts ' ----------- '
+    puts " #{@board[6]} | #{@board[7]} | #{@board[8]} "
+  end
```

Comment:
Please remove these `puts` statements. Instead, return a multiline string and then `puts` the method.

There should be no `puts/print` or `gets` statements outside `bin/main.rb`.